src/client.rs

```diff
@@ -88,7 +153,106 @@ impl Record {
+    #[cfg(feature = "attachments")]
+    pub async fn get_attachment<T, E>(&mut self, client: &mut Client) -> Option<Result<T, ClientError>>
+    where T: TryFrom<Vec<u8>, Error = E>,
+        E: 'static + Send + Sync + std::error::Error
+    {
+        if let Attachment::Pending = self.attachment {
+            let maybe_attachment = self.value.as_object()
+                .and_then(|obj| obj.get("attachment"))
+                .map(|a| serde_json::from_value::<AttachmentMetadata>(a.clone()));
+
+            match maybe_attachment {
+                Some(Ok(meta)) => {
+                    self.attachment = Attachment::Some(meta);
+                }
+                Some(Err(err)) => {
+                    // serde_json::Error does not implement std::error::Error,
+                    // weirdly, so convert it to a string.
+                    return Some(Err(ClientError::AttachmentFormatError(
+                        anyhow!("Could not convert attachment to requested type: {}", err)
+                    )));
+                }
+                None => {
+                    self.attachment = Attachment::None;
+                }
+            }
+        }
+
+        match self.attachment {
+            Attachment::Some(ref metadata) => {
+                Some(Self::fetch_attachment::<T, E>(client, metadata).await)
+            },
+            Attachment::None => Option::None,
+            Attachment::Pending => unreachable!(),
+        }
+    }
+
+    #[cfg(feature = "attachments")]
+    async fn fetch_attachment<T, E>(
+        client: &mut Client,
+        metadata: &AttachmentMetadata
+    ) -> Result<T, ClientError>
+    where T: TryFrom<Vec<u8>, Error = E>,
+          E: 'static + Send + Sync + std::error::Error
+    {
+        let key = format!("attachment:{}", metadata.hash);
+        let bytes = match client.storage.retrieve(&key) {
+            Ok(bytes) => Ok(bytes),
+
+            Err(StorageError::KeyNotFound { .. }) => {
+                let url = {
+                    let server_info = client.server_info().await?;
+                    match &server_info["capabilities"]["attachments"]["base_url"] {
+                        Value::String(s) => Url::parse(s).map_err(|err|
+                            ClientError::APIError(KintoError::URLError(err))
+                        )?,
+                        Value::Null => {
+                            return Err(ClientError::CompatibilityError(
+                                anyhow!("server does not support attachments")
+                            ));
+                        }
+                        _ => {
+                            return Err(ClientError::CompatibilityError(
+                                anyhow!("server did not return a valid attachment base_url")
+                            ));
+                        },
+                    }
+                };
+
+                let response = client.http_client.get(url).await
```

Comment:
Here you need to join the base url from server info with the attachment relative url 
```rust
                let url = format!("{}/{}", base_url, metadata.location);
                let response = client.http_client.get(url).await
```